react/features/youtube-player/middleware.js

```diff
@@ -0,0 +1,166 @@
+        if (localParticipantId === ownerId && [ 'playing', 'pause' ].includes(status)) {
+            const fakeParticipant = state['features/base/participants'].filter(p => p.isFakeParticipant)[0];
+
+            sendShareVideoCommand(fakeParticipant.id, status, conference, localParticipantId, action.time);
+        }
+        break;
+    }
+
+    return next(action);
+});
+
+/**
+ * Set up state change listener to perform maintenance tasks when the conference
+ * is left or failed, e.g. clear messages or close the chat modal if it's left
+ * open.
+ */
+StateListenerRegistry.register(
+    state => getCurrentConference(state),
+    (conference, { dispatch, getState }, previousConference) => {
+        if (conference && conference !== previousConference) {
+            conference.addCommandListener(SHARED_VIDEO,
+                ({ value, attributes }) => {
+                    const localParticipantId = getLocalParticipant(getState()).id;
+
+                    switch (attributes.state) {
+                    case 'start':
+                        dispatch(setSharedVideoOwner(attributes.from));
+                        dispatch(participantJoined({
+                            conference,
+                            id: value,
+                            isFakeParticipant: true,
+                            avatarURL: `https://img.youtube.com/vi/${value}/0.jpg`,
+                            name: 'YouTube'
+                        }));
+                        dispatch(pinParticipant(value));
+                        break;
+                    case 'playing':
+                        if (localParticipantId !== attributes.from) {
+                            dispatch(setSharedVideoStatus('playing', attributes.time));
+                        }
+                        break;
+                    case 'pause':
+                        if (localParticipantId !== attributes.from) {
+                            dispatch(setSharedVideoStatus('pause', attributes.time));
+                        }
+                        break;
+                    case 'stop':
+                        dispatch(setSharedVideoStatus('stop', ''));
+                        dispatch(participantLeft(value, conference));
+                        break;
+                    }
+                }
+            );
+        }
+    });
+
+
+/**
+ * Dispatches shared video status.
+ *
+ * @param {Store} store - The redux store.
+ * @param {string} next - Todo add doc.
+ * @param {string} action - Todo add doc.
+ * @returns {Function}
+ */
+function _toggleSharedVideo(store, next, action) {
+    const state = store.getState();
+    const { ownerId } = state['features/youtube-player'];
+    const localParticipant = getLocalParticipant(state);
+
+    const conference = getCurrentConference(state);
+    const { status } = state['features/youtube-player'];
+    const fakeParticipant = state['features/base/participants'].filter(p => p.isFakeParticipant)[0];
```

Comment:
find the exact one

Reply:
Now that I think about it, we only allow for 1 participant to share a YT video at a time. Which makes sense I guess. But what happens if someone else tries to start sharing another video?